rfc3164/rfc3164.go

```diff
@@ -169,16 +171,21 @@ func (p *Parser) parseTimestamp() (time.Time, error) {
     }
 
     sub = p.buff[p.cursor : tsFmtLen+p.cursor]
-    ts, err = time.ParseInLocation(tsFmt, string(sub), time.Local)
-    if err == nil {
-      /* Set Year on the Timestamp before converting to UTC so that time zone and
-         DST settings (which are year dependent) can be properly assessed in the
-         conversion. */
-      p.fixTimestampIfNeeded(&ts)
-
-      ts = ts.UTC()
-      found = true
-      break
+
+    // Only test the date pattern if the next character is a space (or the end of the message)
+    // This prevents us picking up the first 4 numbers of a hostname as a year
+    if p.cursor+tsFmtLen == p.l || p.buff[p.cursor + tsFmtLen] == ' ' {
```

Comment:
Add an `Rfc3164ParseOptions` struct as an argument to `NewParser`. This could contain flags for any deviations from the RFC that need to be enabled. Tidier than passing a bunch of bools in directly.